### PR TITLE
[13.0] [MIG] web_widget_domain_editor_dialog: Migration to v13.0

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT25342

This module adds the posibility to generate the domain from the dialog how we see on next image.

Please @chienandalu @Yajo con you review this?

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "restructuredtext.confPath": ""
+}
```
Comment (chienandalu) [inline .vscode/settings.json]: check your gitignore settings ;)

File: web_widget_domain_editor_dialog/static/src/js/basic_fields.js [hunk 2]
```diff
@@ -4,31 +4,31 @@
...
+    const core = require("web.core");
+    const basic_fields = require("web.basic_fields");
+    const DomainEditorDialog = require("web_widget_domain_editor_dialog.DomainEditorDialog");
+    const _t = core._t;
 
     basic_fields.FieldDomain.include({
         _onShowSelectionButtonClick: function(event) {
             event.preventDefault();
-            var _this = this;
+            const self = this;
```
Comment (chienandalu) [inline web_widget_domain_editor_dialog/static/src/js/basic_fields.js]: Actually `_this` was more apropiate: https://developer.mozilla.org/en-US/docs/Web/API/Window/self

File: web_widget_domain_editor_dialog/static/src/js/widget_domain_editor_dialog.js [hunk 3]
```diff
@@ -4,65 +4,40 @@
...
-                        get_groupby: function() {
-                            return [];
-                        },
-                        get_domain: function() {
-                            return _this.options.default_domain;
-                        },
+                dynamicFilters: [
+                    {
+                        description: _.str.sprintf(_t("Selected domain")),
+                        domain: JSON.parse(self.options.default_domain),
```
Comment (chienandalu) [inline web_widget_domain_editor_dialog/static/src/js/widget_domain_editor_dialog.js]: This is failing

**Review by chienandalu — CHANGES_REQUESTED** [review]

**Comment by CarlosRoca13** [comment]
@chienandalu All changes done!

**Review by yajo — APPROVED** [review]

**Review by chienandalu — APPROVED** [review]

**Pull request merged**